File: Paperless TeDi/Models/Document.swift
import Foundation

struct Document: Identifiable, Codable, Hashable {
    let id: Int
    var title: String
    var content: String?
    var created: String
    var added: String?
    var correspondent: Int?
    var documentType: Int?
    var archiveSerialNumber: Int?
    var tags: [Int]
    var notes: [Note]?

    enum CodingKeys: String, CodingKey {
        case id, title, content, created, added, correspondent, tags, notes
        case documentType = "document_type"
        case archiveSerialNumber = "archive_serial_number"
    }

    var dateObject: Date? {
        let iso = ISO8601DateFormatter()
        iso.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        if let d = iso.date(from: created) { return d }
        iso.formatOptions = [.withInternetDateTime]
        if let d = iso.date(from: created) { return d }
        let simple = DateFormatter()
        simple.dateFormat = "yyyy-MM-dd"
        return simple.date(from: created)
    }

    var safeNotes: [Note] { notes ?? [] }
}

struct UploadContainer: Identifiable {
    let id = UUID()
    let data: Data
    let filename: String
}

File: android/app/src/main/java/de/tedi/paperless/network/model/Document.kt
package de.tedi.paperless.network.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Document(
    val id: Int,
    val title: String,
    val content: String? = null,
    val created: String,
    val added: String? = null,
    val correspondent: Int? = null,
    @Json(name = "document_type") val documentType: Int? = null,
    @Json(name = "archive_serial_number") val archiveSerialNumber: Int? = null,
    val tags: List<Int> = emptyList()
)
